numba/cuda/tests/cudapy/test_cooperative_groups.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import print_function
+
+import numpy as np
+
+from numba import config, cuda, int32, void
+from numba.cuda.testing import unittest, CUDATestCase, skip_on_cudasim
+
+
+def cc_X_or_above(major, minor):
+    if not config.ENABLE_CUDASIM:
+        cc = cuda.current_context().device.compute_capability
+        return cc >= (major, minor)
+    else:
+        return True
+
+
+def skip_unless_cc_60(fn):
+    return unittest.skipUnless(cc_X_or_above(6, 0), "requires cc >= 6.0")(fn)
+
+
+@cuda.jit
+def this_grid(A):
+    cuda.cg.this_grid()
+    A[0] = 1.0
+
+
+@cuda.jit
+def sync_group(A):
+    g = cuda.cg.this_grid()
+    g.sync()
+    A[0] = 1.0
+
+
+@cuda.jit
+def no_sync(A):
+    A[0] = cuda.grid(1)
+
+
+def sequential_rows(M):
+    # The grid writes rows one at a time. Each thread reads an element from
+    # the previous row written by its "opposite" thread.
+    #
+    # A failure to sync the grid at each row would result in an incorrect
+    # result as some threads could run ahead of threads in other blocks, or
+    # fail to see the update to the previous row from their opposite thread.
+
+    col = cuda.grid(1)
+    g = cuda.cg.this_grid()
+
+    rows = M.shape[0]
+    cols = M.shape[1]
+
+    for row in range(1, rows):
+        opposite = cols - col - 1
+        M[row, col] = M[row - 1, opposite] + 1
+        g.sync()
+
+
+class TestCudaCooperativeGroups(CUDATestCase):
+    @skip_unless_cc_60
+    def test_this_grid(self):
+        A = np.full(1, fill_value=np.nan)
+        this_grid[1, 1](A)
+
+        # Ensure the kernel executed beyond the call to cuda.this_grid()
+        self.assertFalse(np.isnan(A[0]), 'set it to something!')
```

Comment:
```suggestion
        self.assertFalse(np.isnan(A[0]), 'Value erroneously set!')
```

Reply:
I've added a slightly different phrase: "Value not set" - if `A[0]` is a NaN, then it was not set by the kernel.